src/pg_cron.c

```diff
@@ -1664,18 +1664,23 @@ ManageCronTask(CronTask *task, TimestampTz currentTime)
 				break;
 			}
 
-			/* still waiting for job to complete */
-			if (GetBackgroundWorkerPid(&task->handle, &pid) != BGWH_STOPPED)
-				break;
-
 			toc = shm_toc_attach(PG_CRON_MAGIC, dsm_segment_address(task->seg));
 			#if PG_VERSION_NUM < 100000
 				mq = shm_toc_lookup(toc, PG_CRON_KEY_QUEUE);
 			#else
 				mq = shm_toc_lookup(toc, PG_CRON_KEY_QUEUE, false);
 			#endif
 			responseq = shm_mq_attach(mq, task->seg, NULL);
-			GetBgwTaskFeedback(responseq, task);
+
+			/* still waiting for job to complete */
+			if (GetBackgroundWorkerPid(&task->handle, &pid) != BGWH_STOPPED)
+			{
+				GetBgwTaskFeedback(responseq, task, true);
+				shm_mq_detach(responseq);
+				break;
+			}
+
+			GetBgwTaskFeedback(responseq, task, false);
```

Comment:
I guess it's ok that we don't do a shm_mq_detach below because of the dsm_detach?